src/containers/NewPost/NewPost.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+
+import { Segment } from 'semantic-ui-react';
+import TabBar from '../../components/TabBar/TabBar';
+import StartMenu from '../../components/StartMenu/StartMenu';
+import NewPostComponent from '../../components/NewPost/NewPostComponent';
+class NewPost extends Component {
+	state = {
+		tabBarName: localStorage.getItem('username'),
+	}
+
+	tokenMatch = () => {
+		const { validations } = this.state;
+		let tokenStorage = localStorage.getItem('token');
+		if (validations.token != null && validations.token === tokenStorage) {
+			this.changeValidation('signedIn', true);
+		} else {
+			this.changeValidation('signedIn', false);
+		}
+	}
```

Comment:
This `tokenMatch` method is the same exact replica from the `Home` component. Let's abstract this by pushing this method more up and passing as props through each component.